File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.rb
# @param {Integer[]} nums
# @return {Integer[]}
def apply_operations(nums)
  result = []
  nums.each_index do |i|
    if i < nums.size - 1 && nums[i] == nums[i + 1] && nums[i] != 0
      nums[i] = nums[i] * 2
      nums[i + 1] = 0
    end
    result << nums[i] if nums[i] != 0
  end
  result + [0] * (nums.size - result.size)
end

File: 2460-apply-operations-to-an-array/2460-apply-operations-to-an-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var applyOperations = function(nums) {
  let result = [];
  for (let i = 0; i < nums.length; i++) {
    if (i < nums.length - 1 && nums[i] === nums[i + 1] && nums[i] !== 0) {
      nums[i] *= 2;
      nums[i + 1] = 0;
    }
    if (nums[i] !== 0) {
      result.push(nums[i]);
    }
  }
  while (result.length < nums.length) {
    result.push(0);
  }
  return result;
};